apps/staking/app/register/[nodeId]/multi/SubmitMultiTab.tsx

```diff
@@ -0,0 +1,740 @@
+ *
+ * (B) Resolve contract details.
+ *   - if finalized move to (C). (Contract is for a previous instance of this node)
+ *       TODO: in the future investigate contract reuse at this stage and use this logic:
+ *          - if `node_add_timestamp` is not null, move to (E).
+ *             (It is currently in the network, this state should not be possible)
+ *          - if `node_add_timestamp` is null, reset contract and move to (D).
+ *             (Reuse this contract, reset it).
+ *   - if operator contributed, move to (E).
+ *   - if operator not contributed, move to (D).
+ *
+ * (C) Create contract.
+ *   - Create contract wallet interaction.
+ *   - Poll for created contract until found.
+ *     - Go to (B).
+ *
+ * (D) Contribute to contract.
+ *    - Go to (E).
+ *
+ * (E) Success.
+ *    - Show success page.
+ */
+function SubmitMulti({
+  creationParams,
+  stakingParams,
+}: {
+  creationParams: UseCreateOpenNodeContractParams;
+  stakingParams: Omit<UseContributeStakeToOpenNodeParams, 'contractAddress'>;
+}) {
+  const dict = useTranslations('actionModules.registration.submitMulti');
+  const dictShared = useTranslations('actionModules.shared');
+  const { setIsSubmitting, setIsSuccess, changeTab, setIsError, setContract, address } =
+    useRegistrationWizard();
+
+  const {
+    createOpenNodeContract,
+    resetCreateOpenNodeContract,
+    enabled: isCreateNodeEnabled,
+    createNodeContractStatus,
+    createNodeContractErrorMessage,
+    simulateError: contractSimulateError,
+    writeError: contractWriteError,
+    transactionError: contractTransactionError,
+  } = useCreateOpenNodeRegistration(creationParams);
+
+  const isDeployed = createNodeContractStatus === PROGRESS_STATUS.SUCCESS;
+
+  const { setNetworkInfo } = useNetworkStatus();
+
+  const {
+    data,
+    isError: isErrorFetchContractDetails,
+    error: errorFetchContractDetails,
+    isSuccess: isSuccessFetchContractDetails,
+  } = useStakingBackendQueryWithParams(
+    getContributionContractBySnKey,
+    {
+      nodePubKey: creationParams.nodePubKey,
+    },
+    /**
+     * Starts fetching after the contract is deployed.
+     *  Once data is returned by the query {@link networkInfo} is updated with the latest network
+     *  info from the fetch, this contains the last l2 height.
+     *
+     *  After this first fetch the `refetchInterval` computes the next l2 update timestamp and
+     *  schedules a refetch 5s after this timestamp.
+     *  (l2_height_timestamp + {@link BACKEND.L2_TARGET_UPDATE_INTERVAL_SECONDS} )
+     *
+     *  This refetch interval computation continues after each fetch until the contract address is
+     *  available in the query state. Then the interval is disabled and no more refetches are made.
+     */
+    {
+      enabled: isDeployed || !isCreateNodeEnabled,
+      refetchInterval: (query) => {
+        if (getNonFinalizedLatestDeployedContributorContract(query.state.data)) return false;
+
+        const l2HeightTimestamp =
```

Comment:
It's wild that you have to do an in check at every level of keys in the object 😅